lib/wallet/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2884,39 +2832,29 @@ decodeTransaction ctx (ApiT wid) (ApiSerialisedTransaction (ApiT sealed) _) = do
         pp <- liftIO $ NW.currentProtocolParameters (wrk ^. networkLayer)
         policyXPubM <- fmap (fmap fst . eitherToMaybe)
             <$> liftIO . runExceptT $ W.readPolicyPublicKey @_ @s @k @n wrk wid
-        pure
-            ( inputPaths
-            , collateralInputPaths
-            , outputPaths
-            , acct
-            , acctPath
-            , pp
-            , policyXPubM
-            )
-    pure $ ApiDecodedTransaction
-        { id = ApiT txId
-        , fee = maybe (Quantity 0) (Quantity . fromIntegral . unCoin) fee
-        , inputs = map toInp txinsOutsPaths
-        , outputs = map toOut outsPath
-        , collateral = map toInp collateralInsOutsPaths
-        -- TODO: [ADP-1670]
-        , collateralOutputs = ApiAsArray Nothing
-        , withdrawals = map (toWrdl acct) $ Map.assocs withdrawals
-        , mint = toApiAssetMintBurn policyXPubM toMint
-        , burn = toApiAssetMintBurn policyXPubM toBurn
-        , certificates = map (toApiAnyCert acct acctPath) allCerts
-        , depositsTaken =
-            (Quantity . fromIntegral . unCoin . W.stakeKeyDeposit $ pp)
-                <$ filter ourRewardAccountRegistration
-                    (toApiAnyCert acct acctPath <$> allCerts)
-        , depositsReturned =
-            (Quantity . fromIntegral . unCoin . W.stakeKeyDeposit $ pp)
-                <$ filter ourRewardAccountDeregistration
-                    (toApiAnyCert acct acctPath <$> allCerts)
-        , metadata = ApiTxMetadata $ ApiT <$> metadata
-        , scriptValidity = ApiT <$> scriptValidity
-        , validityInterval = interval
-        }
+        let certs = mkApiAnyCertificate acct acctPath <$> allCerts
+        pure $ ApiDecodedTransaction
+            { id = ApiT txId
+            , fee = maybe (Quantity 0) (Quantity . fromIntegral . unCoin) fee
+            , inputs = map toInp inputPaths
+            , outputs = map toOut outputPaths
+            , collateral = map toInp collateralInputPaths
+            -- TODO: [ADP-1670]
+            , collateralOutputs = ApiAsArray Nothing
+            , withdrawals = map (toWrdl acct) $ Map.assocs withdrawals
+            , mint = toApiAssetMintBurn policyXPubM toMint
+            , burn = toApiAssetMintBurn policyXPubM toBurn
+            , certificates = certs
+            , depositsTaken =
```

Comment:
The change in indentation makes the commit very noisy: Every line is affected, but only the line for the `certificates` was really changed. I was able to review this, but for next time, I recommend to make two commits: One preparatory commit that changes the indentation (pure refactoring) / scope, and one commit that introduces the change in logic, i.e. the expression for the `certificates` field.